src/message/mailbox/parsers.rs

```diff
@@ -0,0 +1,322 @@
+    range.extend('a'..='z');
+    range.extend('A'..='Z');
+    range.extend("!#$%&\'*+-/=?^_`{|}~".chars());
+    one_of(range)
+}
+
+// atom            =       [CFWS] 1*atext [CFWS]
+pub fn atom() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    cfws()
+        .or_not()
+        .ignore_then(atext().repeated().at_least(1))
+        .then_ignore(cfws().or_not())
+}
+
+// dot-atom        =       [CFWS] dot-atom-text [CFWS]
+pub fn dot_atom() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    cfws()
+        .or_not()
+        .ignore_then(dot_atom_text())
+        .then_ignore(cfws().or_not())
+}
+
+// dot-atom-text   =       1*atext *("." 1*atext)
+pub fn dot_atom_text() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    atext().repeated().at_least(1).chain(
+        just('.')
+            .chain(atext().repeated().at_least(1))
+            .repeated()
+            .at_least(1)
+            .flatten(),
+    )
+}
+
+// Quoted strings
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.5
+
+// qtext           =       NO-WS-CTL /     ; Non white space controls
+//
+//                         %d33 /          ; The rest of the US-ASCII
+//                         %d35-91 /       ;  characters not including "\"
+//                         %d93-126        ;  or the quote character
+pub fn qtext() -> impl Parser<char, char, Error = Simple<char>> {
+    let mut range: Vec<char> = vec![33 as char];
+    range.extend((35 as char)..=(91 as char));
+    range.extend((93 as char)..=(126 as char));
+    one_of(range)
+}
+
+// qcontent        =       qtext / quoted-pair
+pub fn qcontent() -> impl Parser<char, char, Error = Simple<char>> {
+    choice((qtext(), quoted_pair()))
+}
+
+// quoted-string   =       [CFWS]
+//                         DQUOTE *([FWS] qcontent) [FWS] DQUOTE
+//                         [CFWS]
+pub fn quoted_string() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    cfws()
+        .or_not()
+        .ignore_then(fws().or_not().ignore_then(qcontent()).repeated())
+        .then_ignore(fws().or_not())
+        .delimited_by(just('"').ignored(), just('"').ignored())
+        .collect()
+}
+
+// Miscellaneous tokens
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.6
+
+// word            =       atom / quoted-string
+pub fn word() -> impl Parser<char, Vec<char>, Error = Simple<char>> {
+    choice((atom(), quoted_string()))
+}
+
+// Address Specification
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.4
+
+// mailbox         =       name-addr / addr-spec
+pub fn mailbox() -> impl Parser<char, (Option<String>, (String, String)), Error = Simple<char>> {
+    choice((addr_spec().map(|addr| (None, addr)), name_addr()))
+}
+
+// name-addr       =       [display-name] angle-addr
+pub fn name_addr() -> impl Parser<char, (Option<String>, (String, String)), Error = Simple<char>> {
+    // NOTE: display-name does not follow the RFC here in order to be
+    // more flexible.
```

Comment:
Can you be specific in the comment what extensions we allow?

Reply:
:heavy_check_mark: